### DISCUSSION TITLE
Generate more compatible videos for mobile

### DISCUSSION THREAD

**Discussion by lijunle** [bpatrik/pigallery2#599] [General]
Some of my recent video conversion is causing incompatibility with Safari. Following [this post](https://superuser.com/a/859075/415479), I provide the following options to ffmpeg custom options. (Notice we cannot have spaces around semicolon `;`)

```
-profile:v main;-pix_fmt yuv420p;-movflags faststart;-c:a aac;-ac 2;-b:a 128k
```

I think pigallery2 should be smart and set these options for best compatibility.

**Comment by bpatrik** [comment]
Thank you!

I will test it later ( sorry only in like a month).
I already set some settings, i need to understand if they are compatible.
Also there is an other thread about hardware acceleration.

-- Sorry for being brief, sent from my phone.

**Comment by theyanglu** [comment]
I recently deployed pigallery 2 (latest) via docker on my synology NAS with the intention of showing back my wife's iOS photo and video backups (many in HEIC and Apple friendly videos).  I, too, was having issues with the **converted** videos on mobile -- on iOS (safari) and Android (Firefox), I couldn't get the videos to play.  On the computer, Mac (Firefox), they'd play just fine.  After sticking in the code above by lijunle in the Settings > Under the Hood > Video > Output options, my new batch of videos was able to play on mobile.  I'd recommend making this code the default (after testing of course).
